indexer.py

```diff
@@ -82,6 +89,45 @@ def calc_tvl(self, session, indx, block):
 
         StatsTVL.insert(session, block, timestamp, tvl)
 
+    def calc_tvc(self, session, indx, block):
+        # read list of protocols from local csv file
+        protocols = {}
+        with open("./meta/protocols.csv", newline="") as csv_file:
+            csv_reader = csv.DictReader(csv_file)
+            for row in csv_reader:
+                if "id" in row:
+                    protocols[row["id"]] = row["defi_llama_slug"]
+
+        timestamp = settings.WEB3_WSS.eth.get_block(block)["timestamp"]
+        accumulated_tvc_for_block = 0
+        for protocol_bytes, defi_llama_slug in protocols.items():
+            protocol = Protocol.get(session, protocol_bytes)
+
+            # Given that our datasource is the spreadsheet, the protocol could not be present in the DB yet
+            if not protocol:
+                continue
+
+            protocol_coverages = ProtocolCoverage.get_protocol_coverages(session, protocol.id)
+
+            if not protocol_coverages:
+                continue
+
+            protocol_coverage = protocol_coverages[0]
+
+            # fetch protocol's TVL from DefiLlama
+            response = requests.get("https://api.llama.fi/protocol/" + defi_llama_slug)
```

Comment:
Surround with try/catch and log any errors. Could be that the api service is offline, in this case we should just skip adding an entry.